pkg/storage/permanent/settings.go

```diff
@@ -186,21 +213,36 @@ func (s *Settings) LatestCommitment() *model.Commitment {
 
 	if err != nil {
 		if ierrors.Is(err, kvstore.ErrKeyNotFound) {
-			return model.NewEmptyCommitment(s.LatestAPI())
+			return model.NewEmptyCommitment(s.apiProvider.LatestAPI())
```

Comment:
We need to use `CurrentAPI()` everywhere

Reply:
Yes, now this can work, before we would have a cyclic dependency between `CurrentAPI` and `LatestCommitment`